### PR TITLE
Python script to read Leetcode API to populate Rating MongoDB

### PR DESCRIPTION
**Pull request by RaymundoGranados**
Issues:
- allQuestionsCount does not include split by Language
- submitStats does not include Language and one problem can be submitted multiple times
- recentSubmissionList is limited to 20 submissions. Even if I put limit as 20000 it returns only 20
- Without knowing all the submissions, Weekly, Monthly and Yearly statistics are calculated based on the 20 recent submissions. Useless

The only option is to record daily All, Hard, Medium and Easy counts

### CODE REVIEW

File: Leetcode-statistics/Leetcode-statistics.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        problems_dict[p["stat"]["question__title_slug"]] = difficulty_dict[p["difficulty"]["level"]]
+    
+    return problems_dict
+
+
+# Prepare Problems dictionary
+problems_dict=getAllProblems()
+
+
+myclient = MongoClient("mongodb://54.206.26.87:27017/")
```
Comment (GeorgePlotnikov) [inline Leetcode-statistics/Leetcode-statistics.py:30]: should be in the configuration file

File: Leetcode-statistics/Leetcode-statistics.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+# myclient = MongoClient("mongodb://localhost:27017/")
+mydb = myclient["ratingdb"]
```
Comment (GeorgePlotnikov) [inline Leetcode-statistics/Leetcode-statistics.py:32]: should be in the configuration file

File: Leetcode-statistics/Leetcode-statistics.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+mycol = mydb["users"]
```
Comment (GeorgePlotnikov) [inline Leetcode-statistics/Leetcode-statistics.py:33]: should be in the configuration file

File: Leetcode-statistics/Leetcode-statistics.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+url = "https://leetcode.com/graphql"
```
Comment (GeorgePlotnikov) [inline Leetcode-statistics/Leetcode-statistics.py:35]: should be in the configuration file

File: Leetcode-statistics/Leetcode-statistics.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
+from collections import defaultdict
+from datetime import datetime
+from pymongo import MongoClient
+import requests
+
+def getAllProblems():
+    url = "https://leetcode.com/api/problems/all/"
```
Comment (GeorgePlotnikov) [inline Leetcode-statistics/Leetcode-statistics.py:7]: should be in the configuration file

**Review by GeorgePlotnikov — APPROVED** [review]

File: Leetcode-statistics/Leetcode-statistics.py (hunk 1 above)
Comment (vKeeReaL) [reply Leetcode-statistics/Leetcode-statistics.py:30]: configs/prod.env MONGO_DB_URL
This file creating by action job, you able to use it

Comment (vKeeReaL) [reply Leetcode-statistics/Leetcode-statistics.py:30]: also you can write whatever you want to the action job

**Review by vKeeReaL — APPROVED** [review]

File: Leetcode-statistics/Leetcode-statistics.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            yearly_statistics[days_old//30][(s["lang"],s["difficulty"])]+=1
+
+        language_dict=[]
+        for key,value in today_statistics.items():
+            x,y = key
+            language_dict.append({
+                "language":x,
+                "difficulty": y,
+                "solved": value,
+            })
```
Comment (vKeeReaL) [inline Leetcode-statistics/Leetcode-statistics.py:141]: Could be moved to a separated function to avoid DRY approach.

**Review by vKeeReaL — APPROVED** [review]

File: Leetcode-statistics/Leetcode-statistics.py (hunk 1 above)
Comment (RaymundoGranados) [reply Leetcode-statistics/Leetcode-statistics.py:30]: Where is configs/prod.env?

Comment (GeorgePlotnikov) [reply Leetcode-statistics/Leetcode-statistics.py]: yes

**Pull request merged**